.github/workflows/pytorch.yml

```diff
@@ -76,11 +78,34 @@ jobs:
       - name: Set up Docker
         uses: docker/setup-docker-action@v4
 
+      - name: Create unique cache key from the year and week (YYYY-WW)
+        id: cache_suffix
+        run: echo "week=$(date -u +%G-%V)" >> "$GITHUB_OUTPUT"
+
+      # Restore cache if available. GitHub automatically evicts cache entries that have not been
```

Comment:
Why do we need to do this? ccache will maintain the limit, and we can use `--evict-older-than` if this is insufficient.

Reply:
Hmmm. I wasn't aware of the `--evict-older-than` option but I don't think we want an "evict older than"; we need an "evict UNUSED older than"...

Nonetheless, the GitHub `actions/cache` won't update the cache with the standard workflow without updating the key on a regular basis (see PR description). I think maybe it _is_ better to just maintain a single project cache that we only fill if we're on the `main` branch OR the project (Tool-Solutions) cache is empty

Reply:
Ahh, I understand, cache elements are intended to be immutable and uniquely identified by their key. Have you looked at standard ways of handling ccache in GH actions? E.g. https://github.com/marketplace/actions/ccache-for-gh-actions

Reply:
> cache elements are intended to be immutable and uniquely identified by their key

Precisely

> Have you looked at standard ways of handling ccache in GH actions? E.g. https://github.com/marketplace/actions/ccache-for-gh-actions

It's a great point but I haven't considered it because we have a slightly odd use-case; we're not installing `ccache` just inside our GitHub runner but _inside the container_. On top of needing to do the `ccache` installation inside the container, we also have to handle any `ccache` configuration inside the container. Therefore I'm not sure we get any benefit from the action in our context